### PR TITLE
updated news, also no longer allowing event handlers to raise anything other than a Defect in the Web3 API

### PR DESCRIPTION
**Pull request by onqtam**

### CODE REVIEW

File: beacon_chain/mainchain_monitor.nim [hunk 1]
```diff
@@ -564,6 +565,9 @@ proc run(m: MainchainMonitor, delayBeforeStart: Duration) {.async.} =
 
     except CatchableError as exc:
       warn "Received invalid deposit", err = exc.msg, j
+    except Exception as err:
+      # chronos still raises exceptions which inherit directly from Exception
```
Comment (arnetheduck) [inline beacon_chain/mainchain_monitor.nim:569]: `if err is Defect: raise (ref Defect)(err) else: handle_like_catchable`

**Comment by onqtam** [comment]
I'll probably re-create this PR because the PR in web3 was based off the wrong commit.

File: beacon_chain/mainchain_monitor.nim (hunk 1 above)
Comment (onqtam) [reply beacon_chain/mainchain_monitor.nim:569]: fixed

**Comment by onqtam** [comment]
I just squashed & force-pushed into the same branch so no need for another PR

**Comment by tersec** [comment]
> I just squashed & force-pushed into the same branch so no need for another PR.
> This PR is predicated on these ones:
> 
>     * [status-im/nim-web3#21](https://github.com/status-im/nim-web3/pull/21)
> 
>     * [status-im/nim-json-rpc#75](https://github.com/status-im/nim-json-rpc/pull/75)

Presumably this still shouldn't be merged, as both of those remain open?

**Comment by onqtam** [comment]
@tersec yes - this is not ready for merging yet. I have to think again if only a `Defect` should be allowed outside of the callbacks. I'll merge those when ready.

**Pull request merged**